tests/integration/ui_write/base.py

```diff
@@ -185,6 +185,16 @@ def test_roundtrip(self):
         str(self.container)  # added as a test to make sure printing works
         self.assertNotEqual(id(self.container), id(self.read_container))
         self.assertContainerEqual(self.read_container, self.container)
+        self.validate()
+
+    def validate(self):
+        # validate created file
+        if os.path.exists(self.filename):
+            with NWBHDF5IO(self.filename, mode='r') as io:
+                errors = pynwb_validate(io)
+                if errors:
+                    for err in errors:
+                        raise Exception(err)
 
```

Comment:
This is the core change -- all roundtrip tests will be validated before the file gets deleted. Validation errors become exceptions.